src/fromager/resolver.py

```diff
@@ -243,3 +244,64 @@ def is_satisfied_by(self, requirement: Requirement, candidate: Candidate) -> boo
     def get_dependencies(self, candidate: Candidate) -> list:
         # return candidate.dependencies
         return []
+
+
+class GitHubTagProvider(ExtrasProvider):
+    def __init__(
+        self,
+        organization,
+        repo,
+    ):
+        super().__init__()
+        self.organization = organization
+        self.repo = repo
+        self.client = github.Github()
+
+    def identify(self, requirement_or_candidate):
+        return canonicalize_name(requirement_or_candidate.name)
+
+    def get_extras_for(self, requirement_or_candidate):
+        # Extras is a set, which is not hashable
+        return tuple(sorted(requirement_or_candidate.extras))
+
+    def get_base_requirement(self, candidate):
+        return Requirement(f"{candidate.name}=={candidate.version}")
+
+    def get_preference(self, identifier, resolutions, candidates, information, **kwds):
+        return sum(1 for _ in candidates[identifier])
+
+    def find_matches(self, identifier, requirements, incompatibilities):
+        repo = self.client.get_repo(f"{self.organization}/{self.repo}")
+
+        requirements = list(requirements[identifier])
+        bad_versions = {c.version for c in incompatibilities[identifier]}
+
+        # Need to pass the extras to the search, so they
+        # are added to the candidate at creation - we
+        # treat candidates as immutable once created.
+        candidates = []
+        for tag in repo.get_tags():
```

Comment:
Not every project uses tags that conform to PEP 440. Also I was surprised that packaging's `Version` class can handle `Version("v0.1.0")`. I suggest to make tag name to version more flexible, e.g. with a user-supplied function `fn(tag: str) -> Version`?

Reply:
That would be a good enhancement. I'll file a ticket to follow up with that.

Reply:
#117